File: c/selectionsort.c
#include <stdio.h>
#include <stdlib.h>
void selection_sort(int a[], int n)
{
    int i, j, t, min;
    for (i = 0; i < n - 1; i++)
    {
        min = i;
        for (j = i + 1; j < n; j++)
        {
            if (a[j] < a[min])
            {
                min = j;
            }
        }
        t = a[i];
        a[i] = a[min];
        a[min] = t;
    }
    for (i = 0; i < n; i++)
    {
        printf("%d\t", a[i]);
    }
}

void main()
{
    int a[50], n, i;
    printf("how many numbers in your array : ");
    scanf("%d", &n);
    printf("enter array element : ");
    for (i = 0; i < n; i++)
    {
        scanf("%d", &a[i]);
    }
    selection_sort(a, n);
}

File: FC_Java_InterviewQues/selectionSort.java
package FC_Java_InterviewQues;

public class selectionSort {
	public static int [] sort(int[] a) {
		for(int i=0;i<a.length-1;i++) {
			int min=i;
			for(int j=i+1;j<a.length;j++) {
				if(a[j]<a[min]) {
					min=j;
				}
			}
			int temp=a[min];
			a[min]=a[i];
			a[i]=temp;
			
			
		}
		return a;
	}
	public static void main(String[] args) {
		int a[]= {1,5,3,2,4};
		int c[]=selectionSort.sort(a);
		for(int x:c) {
			System.out.println(x);
		}
	}

	

}
